source/patterns/@aws-solutions-constructs/core/lib/elasticsearch-helper.ts

```diff
@@ -228,48 +227,63 @@ export function buildElasticSearchCWAlarms(scope: Construct): cloudwatch.Alarm[]
+    // Environment agnostic stacks: A ES cluster deploys in 2 AZs with 2 subnets maximum(each subnet in a different AZ).
     // To successfully deploy a ES cluster, construct will need to specify exactly same subnets number as cluster AZs number
     // To summarize: 1. Subnets number must equals to AZs number.   2. Each subnet belongs to different AZ.
     let subnetIds: string[] = vpc.selectSubnets({ onePerAz: true }).subnetIds;
 
     if (props?.elasticsearchClusterConfig) {
-      const disabledZoneAwareness = 'zoneAwarenessEnabled' in props.elasticsearchClusterConfig &&
-        !props.elasticsearchClusterConfig.zoneAwarenessEnabled;
-      const twoAvailabilityZoneConfig = 'zoneAwarenessConfig' in props.elasticsearchClusterConfig &&
-        props.elasticsearchClusterConfig.zoneAwarenessConfig &&
-        'availabilityZoneCount' in props.elasticsearchClusterConfig.zoneAwarenessConfig &&
-        props.elasticsearchClusterConfig.zoneAwarenessConfig.availabilityZoneCount === 2;
+      const disabledZoneAwareness = checkDisabledZoneAwareness(props);
+      const twoAvailabilityZoneConfig = checkTwoAvailabilityZoneConfig(props);
 
-      if (disabledZoneAwareness) { // If zoneAwarenessEnabled set to false, ES cluster deploy in 1 AZ with 1 subnet
-        subnetIds = subnetIds.slice(0, 1);
-      } else if (twoAvailabilityZoneConfig) { // If zoneAwarenessConfig.availabilityZoneCount set to 2, ES cluster deploy in 2 AZ with 2 subnets.
-        subnetIds = subnetIds.slice(0, 2);
+      if (disabledZoneAwareness) { // If zone awareness is disabled, ES cluster deploy in 1 AZ with 1 subnet
+        subnetIds = extractSubnetIdsByLength(subnetIds, 1);
+      } else if (twoAvailabilityZoneConfig) { // If the availabilityZoneCount is set to 2, ES cluster deploy in 2 AZ with 2 subnets(1 subnet per AZ).
+        subnetIds = extractSubnetIdsByLength(subnetIds, 2);
       } else {
-        subnetIds = subnetIds.slice(0, 3);
+        subnetIds = extractSubnetIdsByLength(subnetIds, 3);
       }
     } else {
-      subnetIds = subnetIds.slice(0, 3);
-    }
-
-    const securityGroupIds: string[] = [];
-
-    // Get the associated security group ids otherwise use the default security group ids
-    if (lambdaFunction) {
-      lambdaFunction.connections.securityGroups.forEach(element => securityGroupIds.push(element.securityGroupId));
+      subnetIds = extractSubnetIdsByLength(subnetIds, 3);
     }
 
     return {
       vpcOptions: {
         subnetIds,
-        securityGroupIds,
       },
     };
   }
 
   return {};
+}
+
+function checkDisabledZoneAwareness(props: elasticsearch.CfnDomainProps): boolean {
+  let disabledZoneAwareness: boolean = false;
+
+  if (props.elasticsearchClusterConfig) {
+    disabledZoneAwareness = 'zoneAwarenessEnabled' in props.elasticsearchClusterConfig && !props.elasticsearchClusterConfig.zoneAwarenessEnabled;
```

Comment:
Calling this value disabledZoneAwareness and having true mean it is DISabled, then (apparently) the condition looks for NOT enabled to evaluate to true introduces too many double negatives to be clear.

Is there a reason not to use ZoneAware everywhere? Then if zoneAwarenessEnabled is set, this value is true and we're left without "this value is true if this flag is turned off, so check the flag with a NOT operator so false translates to true" confusion.

Reply:
elasticsearch.ElasticsearchClusterConfigProperty is not an exported member which does not allow us to cast